hail/src/main/scala/is/hail/expr/types/physical/PBaseStruct.scala

```diff
@@ -248,5 +272,75 @@ abstract class PBaseStruct extends PType {
     }
   }
 
+  override def copyFromType(mb: MethodBuilder, region: Code[Region], srcPType: PType, srcStructAddress: Code[Long],
+    allowDowncast: Boolean = false, forceDeep: Boolean = false): Code[Long] = {
+    assert(srcPType.isInstanceOf[PBaseStruct])
+
+    val sourceType = srcPType.asInstanceOf[PBaseStruct]
+
+    if(this.fields == sourceType.fields) {
```

Comment:
I am not checking fundamentalType here, but am below. Primarily because I am still not entirely certain whether elements/fields should be copyable by their fundamentalType identity rather than their own identity.

In the same vein, if we do want to allow fundamentalType identity, would we want a typesFundamental instance attribute on this class?